new fsdp dataset: nvidia-deeplearningexamples

mvinci12 opened the pull request:
*Issue #, if available:*

*Description of changes:*
due to allenai/c4 dataset error `**[rank3]: FileNotFoundError: gzip://c4-validation.00003-of-00008.json::hf://datasets/allenai/c4@1588ec454efa1a09f29cd18ddd04fe05fc8653a2/en/c4-validation.00003-of-00008.json.gz**`, we are changing dataset to https://huggingface.co/datasets/BEE-spoke-data/Nvidia-DeepLearningExamples/viewer/default/train?views%5B%5D=train

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

KeitaW commented:
Internally discussed with @mvinci12 for the motivation, however [Nvidia-DeepLearningExamples](https://github.com/NVIDIA/DeepLearningExamples) seems to be a coding dataset, while [C4 dataset](https://huggingface.co/datasets/allenai/c4) is more general corpus. Can we see if other general corpus, such as [FineWeb](https://huggingface.co/datasets/HuggingFaceFW/fineweb) can be used as an alternative?

mvinci12 commented:
new commit uses wikimedia/wikipedia dataset (https://huggingface.co/datasets/wikimedia/wikipedia). also added new arguments for users:

--train_split : the specific HF split you'd like to use for training
--val_split : split for validation
--n_samples_train : number of samples you'd like to use for training 
--n_samples_val : number of samples you'd like to use for validation

training and validation split were hardcoded to use "train" and "validation" previously, but new popular datasets only have "train" as a split.

KeitaW commented on 3.test_cases/pytorch/FSDP/src/model_utils/train_utils.py:
```diff
@@ -474,10 +474,12 @@ def create_streaming_dataloader(dataset,
+                      n_samples=6150):
     print(f"dataset={dataset}, name={name}")
     tokenizer = AutoTokenizer.from_pretrained(tokenizer)
     data = load_dataset(dataset, name=name, streaming=True, split=split, trust_remote_code=True).shuffle(42+global_rank)
```
Aren't we getting different shuffle per GPU hence loading different splits per rank?

KeitaW commented on 3.test_cases/pytorch/FSDP/src/model_utils/train_utils.py:
```diff
@@ -474,10 +474,12 @@ def create_streaming_dataloader(dataset,
     print(f"dataset={dataset}, name={name}")
     tokenizer = AutoTokenizer.from_pretrained(tokenizer)
     data = load_dataset(dataset, name=name, streaming=True, split=split, trust_remote_code=True).shuffle(42+global_rank)
+    data = data.take(n_samples)
```
Why do we need to sample here? The resultant dataset seems to be very small.

mvinci12 replied:
Yeah, not needed for this dataset. But I'm thinking it should be configurable for larger/custom datasets

KeitaW reviewed (commented):
added comments

mhuguesaws commented:
Run failure due to

> requests.exceptions.HTTPError: 429 Client Error: Too Many Requests for url: https://huggingface.co/datasets/wikimedia/wikipedia/resolve/b04c8d1ceb2f5cd4588862100d08de323dccfbaa/20231101.en/train-00032-of-00041.parquet

Check results in CI https://github.com/aws-samples/awsome-distributed-training/actions/runs/15500434999/job/43646863450

nghtm commented:
@mvinci12 You can find readme instructions for creating HF Token here: https://github.com/aws-samples/awsome-distributed-training/tree/main/3.test_cases/pytorch/picotron#build-environment

mvinci12 commented:
newest commit is back to c4 dataset, and i added an argument to input the huggingface access token. the regression test will probably fail due to no huggingface token in argument but working with @amanshanbhag on this tomorrow

mhuguesaws commented:
HF token was already in the doc for mistral models since mandatory. HF_TOKEN as environment variable is sufficient.

mhuguesaws commented:
Let's keep C4 dataset, since the cause was unauthenticated requested to HF. New PR will probably better and we can close this one. Thoughts ?
